src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java

```diff
@@ -0,0 +1,164 @@
+    ArrowDeserializationSchema<VectorSchemaRoot> arrowDeserializationSchema =
+        new ArrowDeserializationSchema<VectorSchemaRoot>(
+            VectorSchemaRoot.class, jsonStringArrowSchema, null);
+    assertThat(arrowDeserializationSchema).isNotNull();
+  }
+
+  public ReadSession getReadSession() {
+    EnhancedBigQueryReadStub stub = mock(EnhancedBigQueryReadStub.class);
+    BigQueryClient bigQueryClient = mock(BigQueryClient.class);
+    UnaryCallable<CreateReadSessionRequest, ReadSession> createReadSessionCall =
+        mock(UnaryCallable.class);
+    BigQueryReadClient readClient = BigQueryReadClient.create(stub);
+    BigQueryClientFactory bigQueryReadClientFactory = mock(BigQueryClientFactory.class);
+    TableInfo table =
+        TableInfo.newBuilder(
+                TableId.of("a", "b"),
+                StandardTableDefinition.newBuilder()
+                    .setSchema(
+                        com.google.cloud.bigquery.Schema.of(
+                            Field.of("name", StandardSQLTypeName.BOOL)))
+                    .setNumBytes(1L)
+                    .build())
+            .build();
+    TableReadOptions tableReadOptions = TableReadOptions.newBuilder().build();
+    ReadSession readSession =
+        ReadSession.newBuilder().setName("abc").setReadOptions(tableReadOptions).build();
+    CreateReadSessionRequest request =
+        CreateReadSessionRequest.newBuilder().setReadSession(readSession).build();
+    Optional<String> encodedBase =
+        Optional.of(java.util.Base64.getEncoder().encodeToString(request.toByteArray()));
+    ReadSessionCreatorConfig config =
+        new ReadSessionCreatorConfigBuilder().setRequestEncodedBase(encodedBase).build();
+    ReadSessionCreator creator =
+        new ReadSessionCreator(config, bigQueryClient, bigQueryReadClientFactory);
+    when(bigQueryReadClientFactory.getBigQueryReadClient()).thenReturn(readClient);
+    when(bigQueryClient.getTable(any())).thenReturn(table);
+    when(stub.createReadSessionCallable()).thenReturn(createReadSessionCall);
+    creator
+        .create(TableId.of("dataset", "table"), ImmutableList.of("col1", "col2"), Optional.empty())
+        .getReadSession();
+    ArgumentCaptor<CreateReadSessionRequest> requestCaptor =
+        ArgumentCaptor.forClass(CreateReadSessionRequest.class);
+    verify(createReadSessionCall, times(1)).call(requestCaptor.capture());
+    return requestCaptor.getValue().getReadSession();
+  }
+
+  public Schema getSchema(ReadSession session) throws IOException {
+    ArrowSchema arrowSchema = session.getArrowSchema();
+    Schema schema =
+        MessageSerializer.deserializeSchema(
+            new ReadChannel(
+                new ByteArrayReadableSeekableByteChannel(
+                    arrowSchema.getSerializedSchema().toByteArray())));
+    return schema;
+  }
+
+  @SuppressWarnings("unused")
+  private MockDynamicTableContext createContextObject() {
```

Comment:
Please remove this method. Why do we have it here? It is not used anywhere.

Reply:
removed